prisma/schema.prisma

```diff
@@ -0,0 +1,93 @@
+datasource db {
+  provider = "mysql"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+generator dbml {
+  provider = "prisma-dbml-generator"
+}
+
+// -----------------------------------------------------------------------------
+// These tables are required by next-auth
+// https://next-auth.js.org/v3/adapters/prisma#setup
+
+model Account {
+  id                 String    @id @default(cuid())
+  userId             String
+  providerType       String
+  providerId         String
+  providerAccountId  String
+  refreshToken       String?
+  accessToken        String?
+  accessTokenExpires DateTime?
+  createdAt          DateTime  @default(now())
+  updatedAt          DateTime  @updatedAt
+  user               User      @relation(fields: [userId], references: [id])
+
+  @@unique([providerId, providerAccountId])
+}
+
+model Session {
+  id           String   @id @default(cuid())
+  userId       String
+  expires      DateTime
+  sessionToken String   @unique
+  accessToken  String   @unique
+  createdAt    DateTime @default(now())
+  updatedAt    DateTime @updatedAt
+  user         User     @relation(fields: [userId], references: [id])
+}
+
+model User {
```

Comment:
Curious as to why users and accounts have different abstractions - how would users have different accounts instead of different users? Are we doing role assignments in this PR too?

Reply:
> Curious as to why users and accounts have different abstractions

We are `next-auth`  and the docs just give you this schema then you can just plug and play.

> Are we doing role assignments in this PR too?

I forgot to add here. We do the checks in sign up just need to see which callback it makes more sense to add it